spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java

```diff
@@ -85,15 +88,22 @@ public Object setupRequest(String message) {
 		if(StringUtils.hasText(processorProperties.getApplicationName())) {
 			applicationName = processorProperties.getApplicationName();
 		}
+		if(StringUtils.hasText(processorProperties.getEnvironmentProperties())) {
+			properties = parse(processorProperties.getEnvironmentProperties());
+		}
 
 		TaskLaunchRequest request = new TaskLaunchRequest(
 				processorProperties.getUri(),
 				commandLineArgs,
 				properties,
 				deploymentProperties,
 				applicationName);
-
-		return request;
+		Map<String, Object> headerMap = new HashMap<>(message.getHeaders());
+		headerMap.put("contentType", "application/json");
```

Comment:
That's not what everybody expects from us.

The Spring Cloud Stream `2.0` makes it as `application/json` by default anyway on the output side.

The end-user is free to override it during binding configuration. But with this explicit, unconditional I'm afraid you are eliminating the ability to override expected content-type.

Does it make sense for you?

Thanks

Reply:
I'm not sure that is correct.  In the example below :
`http --server.port=9000 | tasklauncher-transform... |taskLauncher-local`
The payload that is emitted from the `tasklauncher-transform` is `text/plain;charset=UTF-8` not `application/json`.  After a discussion with Soby, it seems that SCSt honors the original content type of the message.  But in the case of our transform it always emits a POJO as a payload and the downstream sink will more than likely always be a tasklauncher that expects a `application/json` payload.  So that is why the contentType is updated in the header.
Thoughts?

Reply:
OK! Not sure why it is `text/plain;charset=UTF-8`, but I definitely would like to stay away of the idea that we always have `taskLauncher` afterwards. That's not how things work in the Microservice Universe. 
So, if we have this problem with the `contet-type` header, let's override it then, but would be better to do that via our new: https://github.com/spring-cloud-stream-app-starters/core/blob/master/common/app-starters-postprocessor-common/src/main/java/org/springframework/cloud/stream/app/postprocessor/ContentTypeEnvironmentPostProcessor.java

See usage in this commit: https://github.com/spring-cloud-stream-app-starters/load-generator/commit/4a01f670d53dc425eca2c7c84221c83c6363146a

Again: my point is still to let end-user to override the default one.